daemon/logs.go

```diff
@@ -101,9 +104,11 @@ func (daemon *Daemon) ContainerLogs(name string, config *ContainerLogsConfig) er
 				}
 				cLog = tmp
 			}
+
 			dec := json.NewDecoder(cLog)
 			l := &jsonlog.JSONLog{}
 			for {
+				l.Reset()
```

Comment:
@duglin yes, it's moved because of the `continue`. This way we always reset the object when the loop start again. Otherwise I was gonna do `continue; l.Reset(); } ...` which is ugly. This doesn't affect the existing behavior.